### PR TITLE
Experience 7609: Add AI session id to fetches

### PR DESCRIPTION
**Pull request by stephenkao**
This changeset updates our fetch creation to include a new `x-ms-session-id` header to help us identify users across client side and API requests.  We're just populating it with the App Insights-generated session id -- it should be enough for our purposes so why not piggyback on their logic!

**This is a WIP.**  The remaining TODOs are:
- More manual testing around the site
- Add tests

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Add `REACT_APP_APPINSIGHTS_KEY` and `REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING` environment variables (I'm using a demo environment's variables -- they're publicly available, but Slack me if you need them anyway!)
2. Navigate around and ensure that `x-ms-session-id` is included in the requests to our API

## Changes
Screenshot 2022-12-08 at 10 15 19 AM

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes https://github.com/CDCgov/prime-reportstream/issues/7609

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/src/resources/AuthResource.ts [hunk 1]
```diff
@@ -19,6 +20,7 @@ export default class AuthResource extends Resource {
             ...init,
             headers: {
                 ...init.headers,
+                ...getAppInsightsHeaders(),
```
Comment (stephenkao) [inline frontend-react/src/resources/AuthResource.ts:23]: I realize we're in the process of migrating from old fetch patterns to React Query, but I figured we should get this in in the meantime.  Happy to remove it, though!

Comment (kevinhaube) [reply frontend-react/src/resources/AuthResource.ts:23]: Spot on, good lookin' out!

File: frontend-react/src/TelemetryService.ts [hunk 2]
```diff
@@ -43,3 +43,13 @@ const createTelemetryService = () => {
 };
 
 export const ai = createTelemetryService();
+
+export function getAppInsightsHeaders(): { [key: string]: string } {
+    return {
+        "x-ms-session-id": getAppInsightsSessionId(),
```
Comment (kevinhaube) [inline frontend-react/src/TelemetryService.ts:49]: To test, I think we can make a mock of `getAppInsightsSessionId()` and hijack the value, then test through `msw`'s request object, or just object drilling to find it in the created fetch config.

**Comment by kevinhaube** [comment]
Also worth linking the ticket: closes #7609

**Review by kevinhaube — APPROVED** [review]
I think this is good to go! This PR unblocks #7604 for `Platform` 😄

**Comment by stephenkao** [comment]
Thanks, @kevinhaube!  Will merge when tests pass!

**Pull request merged**